mijia/src/bluetooth/mod.rs

```diff
@@ -406,6 +511,34 @@ impl BluetoothSession {
         Ok(characteristics)
     }
 
+    /// Get a list of all descriptors on the given GATT characteristic.
+    pub async fn get_descriptors(
+        &self,
+        characteristic: &CharacteristicId,
+    ) -> Result<Vec<DescriptorInfo>, BluetoothError> {
+        let characteristic_node = self
+            .characteristic(characteristic)
+            .introspect_parse()
+            .compat()
+            .await?;
+        let mut descriptors = vec![];
+        for subnode in characteristic_node.nodes {
+            let subnode_name = subnode.name.as_ref().unwrap();
+            if subnode_name.starts_with("desc") {
+                let descriptor_id = DescriptorId {
+                    object_path: format!("{}/{}", characteristic.object_path, subnode_name).into(),
+                };
+                let uuid =
+                    Uuid::parse_str(&self.descriptor(&descriptor_id).uuid().compat().await?)?;
+                descriptors.push(DescriptorInfo {
```

Comment:
What does this for loop look like as a .filter_map().collect()?

Reply:
Painful, because of the async function call and error handling.